### PR TITLE
Page headers

### PR DESCRIPTION
**Pull request by elacymerys**

### CODE REVIEW

File: thesis-gui/src/components/auth/sign-in/SignIn.tsx [hunk 1]
```diff
@@ -67,11 +69,9 @@ export const SignIn: React.FC = () => {
 
     return (
         <IonPage>
-            <PageHeader/>
+            <PageHeader name={ PAGE_NAME } condense={ false } />
             <IonContent class="ion-padding">
-                <IonToolbar class="ion-margin-bottom">
-                    <IonTitle size="large">Sign In</IonTitle>
-                </IonToolbar>
+                <PageHeader name={ PAGE_NAME } condense={ true } />
```
Comment (MaksymilianK) [inline thesis-gui/src/components/auth/sign-in/SignIn.tsx]: Does this header do anything? I removed it and nothing on the page changed.

File: thesis-gui/src/components/auth/sign-up/SignUp.tsx [hunk 2]
```diff
@@ -77,11 +79,9 @@ export const SignUp: React.FC = () => {
 
     return (
         <IonPage>
-            <PageHeader/>
+            <PageHeader name={ PAGE_NAME } condense={ false } />
             <IonContent className="ion-padding">
-                <IonToolbar>
-                    <IonTitle size="large">Sign Up</IonTitle>
-                </IonToolbar>
+                <PageHeader name={ PAGE_NAME } condense={ true } />
```
Comment (MaksymilianK) [inline thesis-gui/src/components/auth/sign-up/SignUp.tsx]: Same as above. I cannot see this nested header displayed on the page. What does it look like on your computer?

Comment (MaksymilianK) [reply thesis-gui/src/components/auth/sign-up/SignUp.tsx]: Same comment for other components.

File: thesis-gui/src/components/my-account/MyAccount.tsx [hunk 3]
```diff
@@ -18,45 +23,71 @@ const UserInfo: React.FC = () => {
...
+    );
+}
+
 const Ranking = () => {
-  return (
+    const { user } = useUserContext();
+    const { categories } = useCategoryContext();
+
+    const getCategoryNameWithId = (id: number) => {
+        for (let category of categories) {
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-account/MyAccount.tsx]: Same comment as [here](https://github.com/elacymerys/thesis/pull/28#discussion_r1026456994)

File: thesis-gui/src/components/my-account/MyAccount.tsx [hunk 4]
```diff
@@ -18,45 +23,71 @@ const UserInfo: React.FC = () => {
...
+            if (category.id === id) {
+                return category.name;
+            }
+        }
+        return '';
+    }
+
+    const categoryRanksItems = Object.entries(user!.categoryRanks).filter(([_, rank]) => rank > 0).map(([id, rank]) => {
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-account/MyAccount.tsx]: Currently every user start with rank = 0.5. Is this `filter `necessary?

File: thesis-gui/src/components/my-account/MyAccount.tsx [hunk 5]
```diff
@@ -18,45 +23,71 @@ const UserInfo: React.FC = () => {
...
+        return <CategoryRanksItem
```
Comment (MaksymilianK) [inline thesis-gui/src/components/my-account/MyAccount.tsx]: Unnecessary `return`. Can be simplified:
```
map(([id, rank]) => (
        <CategoryRanksItem
            name={ getCategoryNameWithId(parseInt(id)) }
            rank={ rank }
        />
)
```

**Review by MaksymilianK — CHANGES_REQUESTED** [review]

File: thesis-gui/src/components/my-account/MyAccount.tsx (hunk 4 above)
Comment (elacymerys) [reply thesis-gui/src/components/my-account/MyAccount.tsx]: Is it possible that the rank will go down to 0 after some time playing?

Comment (MaksymilianK) [reply thesis-gui/src/components/my-account/MyAccount.tsx]: No, it should not be possible. It can reach **nearly** 0, but not the 0.0

**Review by MaksymilianK — DISMISSED** [review]

**Review by MaksymilianK — APPROVED** [review]

**Pull request merged**